src/lib.rs

```diff
@@ -2,7 +2,39 @@ pub mod cache;
 pub mod client;
 pub mod config;
 pub mod consensus;
+pub mod kura;
 pub mod model;
-pub mod networking;
-pub mod storage;
+pub mod torii;
 pub mod validation;
+pub mod wsv;
+
+use crate::{
+    config::Configuration, consensus::sumeragi::Sumeragi, kura::Kura, model::block::Blockchain,
+    torii::Torii,
+};
+
+pub struct Iroha {
+    torii: Torii,
+}
+
+impl Iroha {
+    pub async fn new(config: Configuration) -> Result<Self, &'static str> {
+        let kura;
+        if &config.mode == "strict" {
+            if let Ok(strict_kura) = Kura::strict_init().await {
```

Comment:
`kura = Kura::strict_init().await?` will be more concise to use here

Reply:
Definitely